src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -66,14 +66,18 @@ public bool OnUpdateTopology(Topology topology, bool forceUpdate = false)
             var state = _state;
             var stateFailures = state.Failures;
             var serverNodes = state.Nodes;
-            var len = Math.Min(serverNodes.Count, stateFailures.Length);
+            var numOfServers = serverNodes.Count;
+            var len = Math.Min(numOfServers, stateFailures.Length);
             for (var i = 0; i < len; i++)
             {
                 if (serverNodes[i].ClusterTag == nodeTag)
                 {
+                    if (numOfServers == 1)
+                        return (i, serverNodes[i]);
```

Comment:
why does this matter?

Reply:
Cluster with a single node, We don't run the server for the first time then we get an exception.
We want to reach the same node over and over again no? this is the special case we were talking about IIRC